### PR TITLE
refactor(*): cleanup

### PR DESCRIPTION
**Pull request by ttarsi**
- removes the `script/bash` directory, in favor of a `Makefile`
- fix github actions so they run on PRs
- remove comments where they are redundant
- Rename `RollupGreeter` and `GlobalGreeter` to `Greeter` and `GreetingBook`
- fix tests
- remove unnecessary gas limit from `GreetingBook`
- fix usage of `xmsg.sender`
- update readme
- other general cleanup and simplifying

Note:

Currently the `make deploy` command is failing with:

```
failed to read artifact source file for `{/path/to/dir}/hello-world-template/lib/omni/contracts/src/interfaces/IFeeOracle.sol`
```

which is strange because building and testing works and `DeployGreetingBook` has nothing to do with `IFeeOracle`.

### CODE REVIEW

**Review by kevinhalliday — APPROVED** [review]
For the error you're getting - run `forge cache clean` and `forge clean` - I've found forge scripts fail with errors like this when you get rid of old scripts

File: src/Greeter.sol [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import {ConfLevel} from "omni//core/src/libraries/ConfLevel.sol";
+
+import {GreetingBook} from "./GreetingBook.sol";
+
+/**
+ * @title Greeter
+ * @dev This contract is deployed to rollups. It transmits greetings to the GreetingBook on the Omni EVM.
+ */
+contract Greeter is XApp {
+    uint64 public constant DEST_TX_GAS_LIMIT = 120_000;
```
Comment (kevinhalliday) [inline src/Greeter.sol]: Maybe name this something that says that it's for `XCALL_GREET_GAS_LIMIT` or `XGREET_GAS_LIMIT`

File: src/Greeter.sol [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    address public greetingBook;
+
+    constructor(address portal, address _greetingBook) XApp(portal, ConfLevel.Latest) {
+        greetingBook = _greetingBook;
+    }
+
+    function greet(string calldata greeting) external payable {
+        bytes memory data = abi.encodeWithSelector(GreetingBook.greet.selector, msg.sender, greeting);
+
+        uint256 fee = xcall(omni.omniChainId(), greetingBook, data, DEST_TX_GAS_LIMIT);
```
Comment (kevinhalliday) [inline src/Greeter.sol]: If we bump sol versions, we can use `abi,.encodeCall`

File: test/Greeter.t.sol [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+        vm.expectCall(
+            address(portal),
+            abi.encodeWithSignature(
+                "feeFor(uint64,bytes,uint64)",
+                portal.omniChainId(),
+                abi.encodeWithSignature("greet(address,string)", user, greeting),
+                greeter.DEST_TX_GAS_LIMIT()
+            )
+        );
```
Comment (kevinhalliday) [inline test/Greeter.t.sol]: I don't think we need to test that `feeFor` is called - the simpler the tests uses cases the better.

File: test/GreetingBook.t.sol [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        portal.mockXCall(
+            mockSourceChainId,
+            address(portal),
+            address(greetingBook),
+            abi.encodeWithSelector(GreetingBook.greet.selector, user, greeting),
+            gasLimit
+        );
+        GreetingBook.Greeting memory lastGreet;
+        (lastGreet.user, lastGreet.message, lastGreet.sourceChainId, lastGreet.timestamp) = greetingBook.lastGreet();
+        assertEq(lastGreet.message, greeting);
```
Comment (kevinhalliday) [inline test/GreetingBook.t.sol]: Thoughts on `portal.mockXCall` util?

File: test/Greeter.t.sol (hunk 3 above)
Comment (ttarsi) [reply test/Greeter.t.sol]: yeah agreed, it just confuses things

File: test/GreetingBook.t.sol (hunk 4 above)
Comment (ttarsi) [reply test/GreetingBook.t.sol]: I love it

Comment (ttarsi) [reply test/GreetingBook.t.sol]: Only thing that's a little confusing is the sender field. We've set it to the portal here, but that doesn't really make sense, in fact is probably an anti-pattern

Comment (kevinhalliday) [reply test/GreetingBook.t.sol]: This is  the `xmsg.sender` - so it should never be the portal. We should change that here

**Pull request merged**